### PR TITLE
[SMALLFIX] Create uber jar for YARN integration

### PR DESCRIPTION
**Pull request by apc999**
Make integration assembly jar for YARN

### CODE REVIEW

File: assembly/pom.xml [hunk 1]
```diff
@@ -24,6 +24,7 @@
 
   <modules>
     <module>client</module>
+    <module>integration</module>
```
Comment (calvinjia) [inline assembly/pom.xml]: Should these be broken up further? I don't see a case for having Yarn and Mesos in the same assembly jar. Also, currently Yarn is not built by default, will this change work without -PYarn?

File: integration/yarn/bin/alluxio-yarn.sh [hunk 2]
```diff
@@ -60,12 +60,12 @@ tar -C $ALLUXIO_HOME -zcf $ALLUXIO_TARFILE \
   integration/yarn/bin/alluxio-worker-yarn.sh \
   integration/yarn/bin/alluxio-application-master.sh \
 
-JAR_LOCAL=${ALLUXIO_HOME}/assembly/server/target/alluxio-assembly-server-${VERSION}-jar-with-dependencies.jar
+JAR_LOCAL=${ALLUXIO_HOME}/assembly/integration/target/alluxio-assembly-integration-${VERSION}-jar-with-dependencies.jar
```
Comment (jsimsa) [inline integration/yarn/bin/alluxio-yarn.sh]: should mesos have a similar change in its scripts?

File: assembly/pom.xml (hunk 1 above)
Comment (apc999) [inline assembly/pom.xml]: this module removed.

File: integration/yarn/bin/alluxio-yarn.sh (hunk 2 above)
Comment (apc999) [inline integration/yarn/bin/alluxio-yarn.sh]: this PR will focus on YARN change. @aaudiber  will have separate change for mesos

**Comment by apc999** [comment]
@calvinjia  add docfix 	797678c

**Comment by apc999** [comment]
@jsimsa Are you good with the code/doc change?

**Pull request merged**